src/types.ts

```diff
@@ -433,6 +433,9 @@ export interface IPluginsArray<T> {
  */
 export type VirtualPopulateMap = Map<string, any & VirtualOptions>;
 
+
+export type QueryMethod<T extends (...args: any) => any> = (...args: Parameters<T>) => ReturnType<T>;
```

Comment:
@ggurkal if you know what this does, could you add some tsdoc to it?

Reply:
Done